modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -1424,6 +1431,169 @@ bool  TiffEncoder::write( const Mat& img, const std::vector<int>& params)
     return writeLibTiff(img_vec, params);
 }
 
+static void extend_cvtColor( InputArray _src, OutputArray _dst, int code )
```

Comment:
Technically, you can use `cv::split` to decompose the image on planes and cv::merge to assemble them in any order you want. Single piece of code should cover all missing types. In case if, let's say `CV_16S` is mot supported by one if functions, you can cast it it CV_16U, do permutations and cast back.

Reply:
Thank you for your comment! 

 I thought splitting from Src to 3 planes(B/G/R) with `cv::split` and merging 3 planes to Dst with `cv::merge` use more memory/computing resource. 

I referenced OpenCV manual again. I think cv::mixChannels is suitable to this purpose. I think this cost result is acceptable.

(This results are tested under virtual machine)
- src image is 1920x1080, a half of upper is random, a harf of lower is 0-filled. 
- Value is the medium value of 10 trials.
- Time contains create src image, encode to buffer and decode to dst image.
- unis is "mill sec"
- If (b)/(a) is smaller, Copy from Src to Dst is faster than mixChannel.

| Mat Type      | (a)Copy | (b)mixChannel | (b)/(a) |
| -- | --: | --: | --: |
| CV_8SC1  |      91   |           91  |     99% |
| CV_8SC3  |     222   |          231  |     96% |
| CV_8SC4  |     285   |          294  |     97% |
| CV_16SC1 |      96   |           97  |     99% |
| CV_16SC3 |     272   |          279  |     97% |
| CV_16SC4 |     353   |          367  |     96% |
| CV_32SC1 |     125   |          125  |    100% |
| CV_32SC3 |     357   |          367  |     97% |
| CV_32SC4 |     469   |          494  |     95% |
| CV_64FC1 |     129   |          131  |     98% |
| CV_64FC3 |     526   |          546  |     96% |
| CV_64FC4 |     684   |          707  |     97% |